src/main/java/org/clojars/gguthe/phpass/PHPass.java

```diff
@@ -131,14 +143,29 @@ private String gensaltPrivate(byte[] input) {
     public String HashPassword(String password) {
         byte random[] = new byte[6];
         this.randomGen.nextBytes(random);
-        // Unportable hashes (Blowfish, EXT_DES) could be added here, but I won't do this.
+
+        // Use BCrypt to hash passwords
+        String hash = cryptPrivate(password, BCrypt.gensalt(BCRYPT_WORKFACTOR, randomGen));

```

Comment:
for parity with the PHP implementation, we should leave `cryptPrivate` as just the plain MD5 implementation, and move this logic out into `HashPassword`.  See https://github.com/hautelook/phpass/blob/master/src/Hautelook/Phpass/PasswordHash.php#L263